lms/djangoapps/courseware/views/index.py

```diff
@@ -414,6 +415,7 @@ def _create_courseware_context(self, request):
             settings.FEATURES.get('ENABLE_COURSEWARE_SEARCH') or
             (settings.FEATURES.get('ENABLE_COURSEWARE_SEARCH_FOR_COURSE_STAFF') and self.is_staff)
         )
+        staff_access = self.is_staff
```

Comment:
is `self.is_staff` here different than `user.is_staff` below?  If not, then I don't think we'll ever get to the last clause of `show_courseware_mfe_link`:

```
if user.is_staff:
    return True

if staff_access and should_redirect_to_courseware_microfrontend(course_key):
    return True
```

If those are the same, the first will always return `True` before we ever even evaluate the second, right?

Reply:
It's not terribly clear because the attribute name was poorly chosen, but they mean different things. `user.is_staff` means "are you global staff?". But `self.is_staff` here is an attribute that gets set to `has_access(request.user, 'staff', self.course)`, so basically "are you on the staff for this course".

Reply:
👍 Cool - That's kinda what I assumed but wanted to sanity check.